### PR TITLE
Add remote computation for GTE

### PR DESCRIPTION
**Pull request by dsmilkov**

### CODE REVIEW

File: lilac/config.py [hunk 1]
```diff
@@ -62,6 +62,10 @@ class EmbeddingConfig(BaseModel):
 
   path: PathTuple
   embedding: str
+  remote: bool = (
```
Comment (brilee) [inline lilac/config.py:65]: should probably default to False?

**Review by brilee — APPROVED** [review]

File: lilac/config.py (hunk 1 above)
Comment (nsthorat) [reply lilac/config.py:65]: +1 here and elsewhere

File: lilac/data/dataset_duckdb.py [hunk 2]
```diff
@@ -1030,7 +1030,8 @@ def compute_signal(
       self.project_dir,
     )
 
-    signal.setup()
+    if not remote:
```
Comment (nsthorat) [inline lilac/data/dataset_duckdb.py:1033]: i think you should leave to the signal to decide

File: lilac/data/dataset_utils.py [hunk 3]
```diff
@@ -41,9 +41,9 @@
 from ..utils import chunks, is_primitive, log, open_file
 
 # The embedding write chunk sizes keeps the memory pressure lower as we iteratively write to the
-# vector store. Embeddings are float32, taking up 4 bytes, so this results in ~130K * dims of RAM
+# vector store. Embeddings are float32, taking up 4 bytes, so this results in ~1.25MB * dims of RAM
 # pressure.
-EMBEDDINGS_WRITE_CHUNK_SIZE = 32_768
+EMBEDDINGS_WRITE_CHUNK_SIZE = 32_768 * 10
```
Comment (nsthorat) [inline lilac/data/dataset_utils.py:46]: revert this or just type the number without multiplication?

File: lilac/embeddings/gte.py [hunk 4]
```diff
@@ -32,6 +39,7 @@ class GTESmall(TextEmbeddingSignal):
   local_batch_size: ClassVar[int] = SENTENCE_TRANSFORMER_BATCH_SIZE
   local_parallelism: ClassVar[int] = 1
   local_strategy: ClassVar[TaskExecutionType] = 'threads'
+  runs_remote: ClassVar[bool] = True
```
Comment (nsthorat) [inline lilac/embeddings/gte.py:42]: - i think we should rename this to supports_garden for code marketing (and elsewhere, but maybe for now just do it here)

**Review by nsthorat — APPROVED** [review]

File: lilac/config.py (hunk 1 above)
Comment (dsmilkov) [reply lilac/config.py:65]: done

File: lilac/data/dataset_utils.py (hunk 3 above)
Comment (dsmilkov) [reply lilac/data/dataset_utils.py:46]: done

File: lilac/data/dataset_duckdb.py (hunk 2 above)
Comment (dsmilkov) [reply lilac/data/dataset_duckdb.py:1033]: done, added signal.setup_remote()

File: lilac/embeddings/gte.py (hunk 4 above)
Comment (dsmilkov) [reply lilac/embeddings/gte.py:42]: done

**Pull request merged**